shesmu-server/src/main/java/ca/on/oicr/gsi/shesmu/input/nabu/NabuJsonRepository.java

```diff
@@ -21,9 +19,14 @@ public NabuJsonRepository() {
 
 	@Override
 	protected NabuValue convert(ObjectNode node) {
-		final Set<Long> upstream = new TreeSet<>();
-		for (JsonNode value : node.get("upstream")) {
-			upstream.add(value.asLong());
+		final JsonNode qcstatus = node.get("qcstatus");
+		String status;
+		if (qcstatus == null || qcstatus.isNull()) {
+			status = "PENDING";
+		} else if (qcstatus.isBoolean()) {
+			status = qcstatus.asBoolean() ? "PASSED" : "FAILED";
+		} else {
+			status = qcstatus.asText();
```

Comment:
What are the chances that this'll return `true` or `false`?

Reply:
I hope not. The reason for this (and the time handling above) is that the Shesmu `/input/nabu` endpoint is going to output the data in Shesmu's format, where this is a string, and not a Boolean. So, this handles the case when it's coming in that pre-processed format.